examples/energy-management-app/energy-management-common/src/DEMTestEventTriggers.cpp

```diff
@@ -0,0 +1,385 @@
+    sPowerAdjustCapabilityStruct.powerAdjustCapability.SetNonNull(powerAdjustmentList);
+    sPowerAdjustmentCapability.SetNonNull(sPowerAdjustCapabilityStruct);
+
+    CHIP_ERROR err = GetDEMDelegate()->SetPowerAdjustmentCapability(sPowerAdjustmentCapability);
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Support, "SetTestEventTrigger_PowerAdjustment failed %s", chip::ErrorStr(err));
+    }
+}
+
+void SetTestEventTrigger_ClearForecast()
+{
+    sPowerAdjustments[0].minPower    = 0;
+    sPowerAdjustments[0].maxPower    = 0;
+    sPowerAdjustments[0].minDuration = 0;
+    sPowerAdjustments[0].maxDuration = 0;
+
+    DataModel::List<const DeviceEnergyManagement::Structs::PowerAdjustStruct::Type> powerAdjustmentList(sPowerAdjustments, 1);
+
+    sPowerAdjustCapabilityStruct.powerAdjustCapability.SetNonNull(powerAdjustmentList);
+    sPowerAdjustCapabilityStruct.cause = PowerAdjustReasonEnum::kNoAdjustment;
+
+    DataModel::Nullable<DeviceEnergyManagement::Structs::PowerAdjustCapabilityStruct::Type> powerAdjustmentCapabilityStruct(
+        sPowerAdjustCapabilityStruct);
+
+    CHIP_ERROR err = GetDEMDelegate()->SetPowerAdjustmentCapability(powerAdjustmentCapabilityStruct);
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Support, "SetTestEventTrigger_PowerAdjustment failed %s", chip::ErrorStr(err));
+    }
+}
+
+void SetTestEventTrigger_StartTimeAdjustment()
+{
+    ConfigureForecast(2);
+
+    // Get the current forecast ad update the earliestStartTime and latestEndTime
+    sForecastStruct = GetDEMDelegate()->GetForecast().Value();
+
+    uint32_t chipEpoch = 0;
+
+    CHIP_ERROR err = GetEpochTS(chipEpoch);
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Support, "ConfigureForecast_EarliestStartLatestEndTimes could not get time");
+    }
+
+    // planned start time, in UTC, for the entire Forecast.
+    sForecastStruct.startTime = chipEpoch;
+
+    // Set the earliest start time, in UTC, to that before the startTime
+    sForecastStruct.earliestStartTime = Optional<DataModel::Nullable<uint32_t>>{ DataModel::Nullable<uint32_t>{ chipEpoch - 60 } };
+
+    // Planned end time, in UTC, for the entire Forecast.
+    sForecastStruct.endTime = chipEpoch * 3;
+
+    // Latest end time, in UTC, for the entire Forecast which is > sForecastStruct.endTime
+    sForecastStruct.latestEndTime = Optional<uint32_t>(chipEpoch * 3 + 60);
+
+    GetDEMDelegate()->SetForecast(DataModel::MakeNullable(sForecastStruct));
+}
+
+void SetTestEventTrigger_StartTimeAdjustmentClear()
+{
+    // Get the current forecast ad update the earliestStartTime and latestEndTime
+    sForecastStruct = GetDEMDelegate()->GetForecast().Value();
+
+    sForecastStruct.startTime = static_cast<uint32_t>(0);
+    sForecastStruct.endTime   = static_cast<uint32_t>(0);
+
+    sForecastStruct.earliestStartTime = Optional<DataModel::Nullable<uint32_t>>();
+    sForecastStruct.latestEndTime     = Optional<uint32_t>();
```

Comment:
```suggestion
    sForecastStruct.earliestStartTime = NullOptional;
    sForecastStruct.latestEndTime     = NullOptional;
```
just like yesterday.